sensors/triggers/kafka/kafka.go

```diff
@@ -115,12 +115,21 @@ func NewKafkaTrigger(sensor *v1alpha1.Sensor, trigger *v1alpha1.Trigger, kafkaPr
 		kafkaProducers[trigger.Template.Name] = producer
 	}
 
-	return &KafkaTrigger{
+	kafkaTrigger := &KafkaTrigger{
 		Sensor:   sensor,
 		Trigger:  trigger,
 		Producer: producer,
 		Logger:   logger.With(logging.LabelTriggerType, apicommon.KafkaTrigger),
-	}, nil
+	}
+
+	//must read from the Errors() channel or the async producer will deadlock.
+	go func() {
```

Comment:
Will this goroutine exit after `Execute()`?

Reply:
No, this goroutine will run until the sensor pod exit.

Reply:
This block of code should go into the section above line 116, as for now, every time when the trigger executes, we run `NewKafkaTrigger()` - good thing is, we reuse the same producer.

Going forward, we should change it to only run `NewKafkaTrigger()` one time.

Reply:
ok, got it. Thanks for explaining. Done.